src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -32,6 +32,8 @@ public class TagHelperDescriptorFactory
                 RegexOptions.None,
                 Constants.RegexMatchTimeout);
 
+        private static readonly char[] UnsupportedAttributeSelectors = new[] { '~', '|', '*' };
```

Comment:
For attribute selectors how else would you do it? Whitelist every character, every number, `^`, `$`, every other other valid symbol? 

Blacklisting here should be fine. This is purely to notify the user they're doing something that's known not to be implemented. Other option is to simply noop and let the chars be a part of the attribute name. We're not trying to protect the user from shooting themselves in the foot 100% of the time, just 90% of the time.